src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/Nuxtour.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.ui.stepsdesign;
+
+import io.cdap.cdap.ui.utils.Constants;
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+
+public class Nuxtour {
+  private static final String NUXTOUR_MODAL_XPATH =
+    "//div[contains(@class, 'guided-tour-tooltip') and not(contains(@style, 'display: none'))]";
```

Comment:
is there a better way to locate these?

Reply:
is adding data-testid tags an option for these elements? other than that, using css selectors wouldn't look cleaner than what you currently have

Reply:
I think the welcome tour modal uses a weird library, rather than a component file. Seems like I cannot add custom css selectors to it